src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -477,21 +477,21 @@ std::unordered_set<size_t> RuntimeConfigurator::MHAParallelWAOptimizer::find_uns
     return unsqueezed_params;
 }
 
-std::unordered_set<ExpandedLoopInfoPtr> RuntimeConfigurator::MHAParallelWAOptimizer::find_loops_to_split(
+std::vector<ExpandedLoopInfoPtr> RuntimeConfigurator::MHAParallelWAOptimizer::find_loops_to_split(
     const lowered::LinearIRCPtr& linear_ir,
     const std::unordered_set<size_t>& unsqueezed_params) {
     const auto loop_manager = linear_ir->get_loop_manager();
-    std::unordered_set<ExpandedLoopInfoPtr> loops_to_split;
+    std::set<size_t> loop_idces_to_split;
     std::vector<size_t> prev_loop_idces;
 
-    auto add_loops_to_split = [&](const ExpressionPtr& expr) {
+    auto add_loop_idx_to_split = [&](const ExpressionPtr& expr) {
         const auto& loop_idces = expr->get_loop_ids();
         if (loop_idces != prev_loop_idces) {
             prev_loop_idces = loop_idces;
             for (const auto& loop_id : loop_idces) {
                 const auto expanded_loop_info = loop_manager->get_loop_info<ExpandedLoopInfo>(loop_id);
                 if (expanded_loop_info->get_dim_idx() == m_dim_idx) {
-                    loops_to_split.insert(expanded_loop_info);
+                    loop_idces_to_split.insert(loop_id);
```

Comment:
By the way, one more idea is to contain `ExpandednLoopInfo` in order of execution numbers:
you can put loops to the `unordered_set` as before since we have to have only unique loops.
Then on L511 you can create the vector from this set sorting by `exec_num` of the latest output port (the latest output port is the `LoopEnd`). Thus, we will get sorted vector of loops by execution order.
Probably it's better idea than being based on a decision that all `loop_ids` are sorted by execution order. If we remove this logic, we will have to rewrite the logic here too. I hope that `exec_num` will be always implemented
What do you think?

Reply:
Discussed offline: it was agreed to leave the current logic as is, since the common `update_loop_info` logic relies on the sorted `loop_ids` by exec order as well.